### ISSUE TITLE
Request: add option for no-fallthrough with newlines

### ISSUE THREAD

**Issue by SheetJSDev** [eslint/eslint#15703]
### Environment

Node version: v16.14.0
npm version: v8.3.1
Local ESLint version: v8.11.0 (Currently used)
Global ESLint version: v7.23.0
Operating System: darwin 21.1.0

### What parser are you using?

Default (Espree)

### What did you do?

```js
var x = 3;
switch(x){
  case 0:
  case 1: break;

  case 2:
    
  case 3:
    break;
}
```

[ESLint Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHggPSAzO1xuc3dpdGNoKHgpe1xuICBjYXNlIDA6XG4gIGNhc2UgMTogYnJlYWs7XG5cbiAgY2FzZSAyOlxuICAgIFxuICBjYXNlIDM6XG4gICAgYnJlYWs7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZ2V0dGVyLXJldHVybiI6Miwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNsYXNzLWFzc2lnbiI6Miwibm8tY29tcGFyZS1uZWctemVybyI6Miwibm8tY29uZC1hc3NpZ24iOjIsIm5vLWNvbnN0LWFzc2lnbiI6Miwibm8tY29uc3RhbnQtY29uZGl0aW9uIjoyLCJuby1jb250cm9sLXJlZ2V4IjoyLCJuby1kZWJ1Z2dlciI6Miwibm8tZGVsZXRlLXZhciI6Miwibm8tZHVwZS1hcmdzIjoyLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOjIsIm5vLWR1cGUtZWxzZS1pZiI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW1wb3J0LWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjoyLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNldHRlci1yZXR1cm4iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW5zYWZlLW9wdGlvbmFsLWNoYWluaW5nIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjoyLCJuby11c2VsZXNzLWNhdGNoIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8td2l0aCI6MiwicmVxdWlyZS15aWVsZCI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC10eXBlb2YiOjJ9LCJlbnYiOnt9fX0=)

### What did you expect to happen?

No errors

### What actually happened?

```
  8:3  error  Expected a 'break' statement before 'case'  no-fallthrough
```

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

It might also make sense to support comments:

```js
switch(x) {
  // currently accepted
  case 0:
  case 1:
    break;

  // this issue
  case 2:

  case 3:
    break;

  // general proposal
  /* a note about 4 */
  case 4:
  /* a note about 5 */
  case 5:
    break;
}
```

**Comment by nzakas** [comment]
Thanks for the report. I’m curious why you believe this is a bug? ESLint is reporting that there should be a break before `case 3`, which is correct. Because of the extra beeline after `case 2`, ESLint doesn’t know if you intend for it to fall through or not.

**Comment by SheetJSDev** [comment]
https://eslint.org/docs/rules/no-fallthrough does not actually define "unintentional fallthrough", but each of the bad examples look like

```js
switch(x) {
  case 1:
    // ...
    doSomething(); // there is at least one statement 

  case 2:
    // ...
}
```

There is no example in the docs page with zero statements in a case clause and blank lines before the next statement, so this question focuses on an undocumented corner case.

IMHO "unintentional fallthrough" requires at least one statement in the case clause.

The spec does not use the phrase "fall through", but other languages do have definitions for this.  C++17 / C23 formalized the "fallthrough attribute" and GCC/Clang support `-Wimplicit-fallthrough`.  For example:

```C++
int main() {
  int i = 0;
  switch(3) {
    case 0:
    case 1:
      break;

    /* one blank line */
    case 2:

    case 3:
      break;

    /* two blank lines */
    case 4:


    case 5:
      break;

    /* one statement */
    case 6:

      i = 1;

    case 7: break;
  }
  return 0;
}
```

Running this through `clang` only shows a warning in the last part (one statement):

```zsh
% clang++ -Wimplicit-fallthrough test.cc
test.cc:26:5: warning: unannotated fall-through between switch labels [-Wimplicit-fallthrough]
    case 7: break;
    ^
test.cc:26:5: note: insert 'break;' to avoid fall-through
    case 7: break;
    ^
    break; 
```

**Comment by LvChengbin** [comment]
I encountered the same problem after I added comments before a `case` statement.

```ts
switch( true ) {
    case 'a' :
    case 'b' :
}


switch( true ) {
    case 'a' :
    /**
     * 
     */
    case 'b' :
}
```

And you can see the demo here: https://eslint.org/demo#eyJ0ZXh0Ijoic3dpdGNoKCB0cnVlICkge1xuY2FzZSAnYScgOlxuY2FzZSAnYicgOlxuXHRcbn1cblxuXG5zd2l0Y2goIHRydWUgKSB7XG5jYXNlICdhJyA6XG4vKipcbiAqIFxuICovXG5jYXNlICdiJyA6XG5cdFxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyLCJpbmRlbnQiOjJ9LCJlbnYiOnt9fX0=

**Comment by nzakas** [comment]
In all of the cases mentioned, the rule is behaving as expected. Only cases without empty lines or comments between them are considered valid. As soon as they are separated, this will trigger the warning.

We can update the docs to show these examples.

**Comment by thw0rted** [comment]
Just for trackback, there are bunch of old locked issues about inconsistent no-fallthrough behavior:

https://github.com/eslint/eslint/issues/6741
https://github.com/eslint/eslint/issues/7889
https://github.com/eslint/eslint/issues/9080
https://github.com/eslint/eslint/issues/9559
https://github.com/eslint/eslint/issues/10608
https://github.com/eslint/eslint/issues/13339
https://github.com/eslint/eslint/issues/14701 (finally actually got implemented)

I came here looking for a way to handle `commentPattern` where it isn't the last line immediately before the next `case`, and found there's a closed PR to implement this:

https://github.com/eslint/eslint/pull/11186

IMHO, it would be an objective improvement to the rule to strictly define fallthrough as a `case` that executes at least one statement, then continues execution to a subsequent `case`, without a matching comment between the last executed line under the first `case` and the first executed line under the next `case`.  This would immediately address all the above linked issues by implementing the behavior they have requested.  The current behavior is surprising (and thus, "bad") given a plain-English understanding of the meaning of "fallthrough".

**Comment by TriMoon** [comment]
It's indeed a bit weird that it only reacts on empty lines, see demos [with](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHggPSAzO1xuc3dpdGNoKHgpe1xuICBjYXNlIDA6IC8vIGZhbGwtdGhvdWdoXG4gIGNhc2UgMTpcbiAgICBicmVhaztcblxuICBjYXNlIDI6IC8vIGZhbGwtdGhvdWdoXG5cbiAgY2FzZSAzOlxuICAgIGJyZWFrO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOjIsIm5vLW9iai1jYWxscyI6Miwibm8tb2N0YWwiOjIsIm5vLXByb3RvdHlwZS1idWlsdGlucyI6Miwibm8tcmVkZWNsYXJlIjoyLCJuby1yZWdleC1zcGFjZXMiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZXR0ZXItcmV0dXJuIjoyLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6Miwibm8tc3BhcnNlLWFycmF5cyI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXVuZGVmIjoyLCJuby11bmV4cGVjdGVkLW11bHRpbGluZSI6Miwibm8tdW5yZWFjaGFibGUiOjIsIm5vLXVuc2FmZS1maW5hbGx5IjoyLCJuby11bnNhZmUtbmVnYXRpb24iOjIsIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyfSwiZW52Ijp7ImJyb3dzZXIiOnRydWUsImVzMjAyMSI6dHJ1ZX19fQ==) and [without](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHggPSAzO1xuc3dpdGNoKHgpe1xuICBjYXNlIDA6IC8vIGZhbGwtdGhvdWdoXG4gIGNhc2UgMTpcbiAgICBicmVhaztcblxuICBjYXNlIDI6IC8vIGZhbGwtdGhvdWdoXG4gIGNhc2UgMzpcbiAgICBicmVhaztcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbG9zcy1vZi1wcmVjaXNpb24iOjIsIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOjIsIm5vLXVudXNlZC1sYWJlbHMiOjIsIm5vLXVudXNlZC12YXJzIjoyLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6eyJicm93c2VyIjp0cnVlLCJlczIwMjEiOnRydWV9fX0=) a blank line in between. _(where only line 8 is deleted)_

**Comment by reviewher** [comment]
Coming from other programming languages, ESLint's current definition of fallthrough is inconsistent.  @SheetJSDev talked about C/C++.  Other languages:

_PHP_: PSR2 guidelines define fallthrough as requiring at least one statement.  This is the same as C/C++

_Java_: Sun style defines a fallthrough as "doesn't include a break statement", so technically you always need a fallthrough comment between cases, no matter how much whitespace is in between.

_Go_: forced breaks at the end of each case clause, even if there is no whitespace.  Special keyword `fallthrough` to fall through.

_Perl_: `given` / `when` statements do not fall through.  Special keyword `continue` to fall through.

_C#_: compilation error if there is no break between case clauses, irrespective of the number of lines.  Special statement `goto case ____` to fall through

_Ruby_: `case` / `when` statements do not fall through

_Rust_: `match` does not have fallthrough support.

_Pascal_: `case` statement has no fallthrough

In summary, "everything but eslint" defines fallthrough in one of two ways:

"intrinsic": each case is separate.  `case a: case b:` are seen as two distinct parts irrespective of the amount of whitespace and newlines between them.

"statement": at least one statement between case clauses with no final break.

ESLint's behavior is surprising and feels inconsistent.

**Comment by divinity76** [comment]
i hope there's a way to tell the linter that the passthrough is intentional tho? for example in the Eclipse Java linter, 
```java
	case FIOC_READ:
		is_read = 1;
	case FIOC_WRITE:
		fioc_do_rw(req, arg, in_buf, in_bufsz, out_bufsz, is_read);
		break;
```
will cause a linter warning, but
```java
	case FIOC_READ:
		is_read = 1;
		/* no break */
	case FIOC_WRITE:
		fioc_do_rw(req, arg, in_buf, in_bufsz, out_bufsz, is_read);
		break;
```
tells the linter that the passthru is intentional

**Comment by nzakas** [comment]
@divinity76 please check out the documentation for the rule:
https://eslint.org/docs/rules/no-fallthrough

**Comment by reviewher** [comment]
@LvChengbin Here's a truly funny workaround that passes eslint:

```js
switch( true ) {
  case 'a' /*

    Put a message here

  */:
  case 'b' : break;

  case 'c':
  case /*

    Put a message here

  */ 'd':
}
```

Both workarounds pass because eslint does not check whitespace between `case`, the value of the case clause, and the colon.

Is this more readable / less bug prone?  IMHO no but it at least passes the rule check

**Comment by mdjermanovic** [comment]
This works as intended. The rule considers only cases that appear in consecutive lines as an intentional fallthrough.

I don't think we should change the default behavior because we don't know if most users would really prefer the suggested one.

However, since there are different opinions on what looks and what does not look like intentional fallthrough, I wouldn't mind adding an option to change the behavior. The new option, when enabled, would allow empty cases regardless of blank lines and comments inside.

**Comment by thw0rted** [comment]
Linters are always going to wind up favoring one style or construct over another.  I agree that knowing what "most users would prefer" sounds useful, but also difficult to achieve in practice.  How does the team make decisions like this, in the absence of a poll of X number of ES programmers?  Maybe the standard should be like Wikipedia, "no original research" -- if you want to make an assertion (like "fallthrough should only include case-statements on consecutive lines", or "fallthrough can include whitespace"), you need to bring a citation from a reputable source backing it up. 

A few posts back, @reviewher did a pretty good job of covering how other languages define fallthrough, which I think makes a good case for making whitespace-counts as default behavior.  Has anyone put together a counter-argument in favor of the current (no-whitespace) behavior, perhaps when the rule was originally added?

**Comment by reviewher** [comment]
I genuinely tried to find an example outside of ESLint and came up short.  Even in JS land:

JSHint takes the "statement" interpretation (you can verify on https://jshint.com/ )

JSLint takes the "statement" interpretation (you can verify on https://www.jslint.com/ )

ESLint "blazing a new trail" is not inherently bad, but there arguably should be a very compelling reason.

**Comment by SheetJSDev** [comment]
@mdjermanovic "most users" probably never use switch statements, and "most switch users" probably never use the fallthrough behavior intentionally (commented or otherwise).  The users who do use switch statements liberally in JS (the relevant demographic for the rule) likely have deep experience with system languages like C and expect the concept to work in the same way.

That said, as discussed in this thread, ESLint stands alone in the current behavior.  IMHO the current behavior encourages unreadable patterns.

**Comment by nzakas** [comment]
The team is in agreement that we can add a new option for this behavior but we won’t be changing the default behavior. Anyone is free to submit a pull request for this change.

**Comment by Gautam-Arora24** [comment]
Can I work on this issue?

**Comment by nzakas** [comment]
@Gautam-Arora24 go for it 👍

**Comment by amareshsm** [comment]
I am interested to work on this issue and i am looking into it.

**Comment by Gautam-Arora24** [comment]
@amareshsm Go for it :)

**Comment by snitin315** [comment]
@amareshsm Can I finish this if you are not working on it actively?

**Comment by amareshsm** [comment]
@snitin315 Yes, I will complete this today.
